### PR TITLE
Create Codebase Structure

### PR DESCRIPTION
**Pull request by HarryHyman**
Create 3 projects:
- **Ffyrn.MonoEngine** - project for the main library
- **Ffyrn.MonoEngine.Example** - project for an example game which will make use of the library, using MonoGame cross-platform OpenGL preset
- **Ffyrn.MonoEngine.Tests** - project to contain the unit tests

Changed the name of the project in the readme.
Installed xUnit within `Ffyrn.MonoEngine.Tests` for unit testing.

### CODE REVIEW

File: Ffyrn.MonoEngine.Example/.config/dotnet-tools.json
Comment (SleeplessOrphan) [inline Ffyrn.MonoEngine.Example/.config/dotnet-tools.json:1]: Trust in the Harry

File: README.md [hunk 1]
```diff
@@ -1 +1,3 @@
-# GameEngine
\ No newline at end of file
+# GameEngine
+
+will be posting project files here then making a pull req later
```
Comment (Eliiia) [inline README.md]: I assume this was not intended to make it into the PR

File: README.md [hunk 2]
```diff
@@ -1 +1,3 @@
-# GameEngine
\ No newline at end of file
+# GameEngine
```
Comment (Eliiia) [inline README.md]: In a different PR, Kirsten changed this to `ffyrn.gameengine` and in an earlier commit Harry changed it to `Ffyrn.MonoEngine`. What are we deciding on?

File: Ffyrn.MonoEngine/Class1.cs [hunk 3]
```diff
@@ -0,0 +1,7 @@
+﻿namespace Ffyrn.MonoEngine
+{
+    public class Class1
```
Comment (Eliiia) [inline Ffyrn.MonoEngine/Class1.cs]: Rename this to `Game` or something similar, add something basic, and implement that into the example game please !!! 
What is on the main branch has to be runnable at all times, and at the moment there is no code to run lol

File: Ffyrn.MonoEngine.Example/app.manifest [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+        <requestedExecutionLevel  level="asInvoker" uiAccess="false" />
+      </requestedPrivileges>
+    </security>
+  </trustInfo>
+
+  <compatibility xmlns="urn:schemas-microsoft-com:compatibility.v1">
+    <application>
+      <!-- A list of the Windows versions that this application has been tested on and is
+           is designed to work with. Uncomment the appropriate elements and Windows will 
+           automatically selected the most compatible environment. -->
```
Comment (Eliiia) [inline Ffyrn.MonoEngine.Example/app.manifest:16]: ARE we testing this on windows vista, 7, 8, 8.1, and 10?

**Review by Eliiia — CHANGES_REQUESTED** [review]

File: Ffyrn.MonoEngine/Class1.cs (hunk 3 above)
Comment (Eliiia) [reply Ffyrn.MonoEngine/Class1.cs]: By "something simple" I do mean something simple, don't go implementing a feature. Display a square on a monogame window or something!

File: README.md (hunk 1 above)
Comment (HarryHyman) [reply README.md]: It was not, it was replaced in my later commit

File: README.md (hunk 2 above)
Comment (HarryHyman) [reply README.md]: `Ffyrn.MonoEngine` would be consistent with the name of the project & namespace and I believe is the name we agreed on. It should be in PascalCase to follow the .NET naming conventions

File: Ffyrn.MonoEngine/Class1.cs (hunk 3 above)
Comment (HarryHyman) [reply Ffyrn.MonoEngine/Class1.cs]: Added a simple `DrawRectangle` method in b03663e.

I also tried to add unit tests but it seems we can't mock `GraphicsDevice` easily, I have left it in commented out as an example though.

File: Ffyrn.MonoEngine.Example/app.manifest (hunk 4 above)
Comment (HarryHyman) [reply Ffyrn.MonoEngine.Example/app.manifest:16]: It seems like this is for windows compatibility mode to have an idea of which OS the program supports and should try, for when future versions of windows come out. I've removed the older ones in 3d38166 and mentioned that win 10 and 11 share the same guid

**Review by SleeplessOrphan — APPROVED** [review]

**Review by Eliiia — APPROVED** [review]
yay

**Pull request merged**